framework/tlvf/src/src/CmduMessageRx.cpp

```diff
@@ -7,28 +7,176 @@
  */
 
 #include <tlvf/CmduMessageRx.h>
-#include <tlvf/CmduParser.h>
-#include <tlvf/CmduTlvParser.h>
+#include <tlvf/WSC/AttrList.h>
+#include <tlvf/ieee_1905_1/tlv1905NeighborDevice.h>
+#include <tlvf/ieee_1905_1/tlvAlMacAddressType.h>
+#include <tlvf/ieee_1905_1/tlvAutoconfigFreqBand.h>
+#include <tlvf/ieee_1905_1/tlvDeviceBridgingCapability.h>
+#include <tlvf/ieee_1905_1/tlvDeviceInformation.h>
+#include <tlvf/ieee_1905_1/tlvEndOfMessage.h>
+#include <tlvf/ieee_1905_1/tlvLinkMetricQuery.h>
+#include <tlvf/ieee_1905_1/tlvLinkMetricResultCode.h>
+#include <tlvf/ieee_1905_1/tlvMacAddress.h>
+#include <tlvf/ieee_1905_1/tlvNon1905neighborDeviceList.h>
+#include <tlvf/ieee_1905_1/tlvPushButtonEventNotification.h>
+#include <tlvf/ieee_1905_1/tlvPushButtonJoinNotification.h>
+#include <tlvf/ieee_1905_1/tlvReceiverLinkMetric.h>
+#include <tlvf/ieee_1905_1/tlvSearchedRole.h>
+#include <tlvf/ieee_1905_1/tlvSupportedFreqBand.h>
+#include <tlvf/ieee_1905_1/tlvSupportedRole.h>
+#include <tlvf/ieee_1905_1/tlvTransmitterLinkMetric.h>
+#include <tlvf/ieee_1905_1/tlvUnknown.h>
+#include <tlvf/ieee_1905_1/tlvVendorSpecific.h>
+#include <tlvf/ieee_1905_1/tlvWsc.h>
+#include <tlvf/wfa_map/tlvApCapability.h>
+#include <tlvf/wfa_map/tlvApMetricQuery.h>
+#include <tlvf/wfa_map/tlvApRadioBasicCapabilities.h>
+#include <tlvf/wfa_map/tlvApRadioIdentifier.h>
+#include <tlvf/wfa_map/tlvChannelPreference.h>
+#include <tlvf/wfa_map/tlvChannelSelectionResponse.h>
+#include <tlvf/wfa_map/tlvClientAssociationControlRequest.h>
+#include <tlvf/wfa_map/tlvClientAssociationEvent.h>
+#include <tlvf/wfa_map/tlvHigherLayerData.h>
+#include <tlvf/wfa_map/tlvOperatingChannelReport.h>
+#include <tlvf/wfa_map/tlvRadioOperationRestriction.h>
+#include <tlvf/wfa_map/tlvSearchedService.h>
+#include <tlvf/wfa_map/tlvSteeringBTMReport.h>
+#include <tlvf/wfa_map/tlvSteeringRequest.h>
+#include <tlvf/wfa_map/tlvSupportedService.h>
+#include <tlvf/wfa_map/tlvTransmitPowerLimit.h>
+
+#include <iostream>
 
 using namespace ieee1905_1;
 
-CmduMessageRx::CmduMessageRx(uint8_t *buff, size_t buff_len) : CmduMessage(buff, buff_len)
+std::shared_ptr<BaseClass> CmduMessageRx::parseNextTlv()
```

Comment:
This change goes a bit against the automatically generated parser that @Galvien still has lined up AFAIU. A step in the right direction would be to make this an external function that takes the ClassList as an argument and that open-codes getNextTlvType, because that's what the generated code would look like.

Nitpick.

Reply:
I don't think we need an automatically generated parser anymore. We can add the missing TLVs for R2 manually, it shouldn't take this long - how many are there?

Reply:
Hehe. There are about 30 new TLVs in R2... I agree though that the effort of maintaining that manually is pretty small. I just thought, since @Galvien has the code anyway, we could easily merge it.

Reply:
The code needs to be reviewed again and added to TLVF so it will take Gal at least half a day I assume, and I rather he work on KW issues for now. We can use his branch to regenerate the R2 TLVs when the time comes, but if someone has the time and will to do that, I won't object to merging the automatic parser, its just its really nice to have and not a must.

Reply:
Right now the code serves more as a method of quickly generating parsing code later to be copied and integrated manually into the codebase rather than a mechanism for creating autogenerated parser files.
So the course of action regarding adding new automatic parsers would be running TLVF once with the required configuration for the autogenerator, then copying it's output manually, the whole process should take 5-10 minutes instead of the hours needed for integrating it fully (and the far more difficult aspect - properly) into TLVF, that's at least where it stands ATM...